pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -542,48 +470,61 @@ func configureGit(d *deploymentSettingsCommandDependencies, gitSSHPrivateKeyPath
 func configureGitHubRepo(d *deploymentSettingsCommandDependencies, vcsInfo *gitutil.VCSInfo) error {
 	sd := d.Deployment
 
-	sd.DeploymentSettings.GitHub = &apitype.DeploymentSettingsGitHub{
-		Repository: fmt.Sprintf("%s/%s", vcsInfo.Owner, vcsInfo.Repo),
+	if sd.DeploymentSettings.GitHub == nil {
+		sd.DeploymentSettings.GitHub = &apitype.DeploymentSettingsGitHub{}
 	}
 
-	options := promptUserMultiSkippable(
-		d.Yes,
-		"GitHub configuration",
-		[]string{
+	sd.DeploymentSettings.GitHub.Repository = fmt.Sprintf("%s/%s", vcsInfo.Owner, vcsInfo.Repo)
+
+	var defaults []string
+
+	if sd.DeploymentSettings.GitHub.PreviewPullRequests {
+		defaults = append(defaults, optPreviewPr)
+	}
+
+	if sd.DeploymentSettings.GitHub.DeployCommits {
+		defaults = append(defaults, optUpdatePushes)
+	}
+
+	if sd.DeploymentSettings.GitHub.PullRequestTemplate {
+		defaults = append(defaults, optPrTemplate)
+	}
+
+	if len(defaults) == 0 {
+		defaults = []string{
 			optPreviewPr,
 			optUpdatePushes,
-			optPrTemplate,
-		},
+		}
+	}
+
+	// For non interactive execution, it automatically accepts the default values
+	options := d.Prompts.PromptUserMultiSkippable(
+		!d.Interactive,
+		"GitHub configuration",
 		[]string{
 			optPreviewPr,
 			optUpdatePushes,
+			optPrTemplate,
 		},
+		defaults,
 		d.DisplayOptions.Color)
 
-	if slices.Contains(options, optPreviewPr) {
-		sd.DeploymentSettings.GitHub.PreviewPullRequests = true
-	}
-
-	if slices.Contains(options, optUpdatePushes) {
-		sd.DeploymentSettings.GitHub.DeployCommits = true
-	}
-
-	if slices.Contains(options, optPrTemplate) {
-		sd.DeploymentSettings.GitHub.PullRequestTemplate = true
-	}
+	sd.DeploymentSettings.GitHub.PreviewPullRequests = slices.Contains(options, optPreviewPr)
+	sd.DeploymentSettings.GitHub.DeployCommits = slices.Contains(options, optUpdatePushes)
+	sd.DeploymentSettings.GitHub.PullRequestTemplate = slices.Contains(options, optPrTemplate)
 
 	return nil
 }
 
 func configureBareGitRepo(d *deploymentSettingsCommandDependencies,
-	remoteURL string, gitSSHPrivateKeyPath string,
+	remoteURL string, gitSSHPrivateKeyPath string, gitSSHPrivateKeyValue string,
 ) error {
 	sd := d.Deployment
 
 	sd.DeploymentSettings.SourceContext.Git.RepoURL = remoteURL
 
-	option := promptUserSkippable(
-		d.Yes,
+	option := d.Prompts.PromptUserSkippable(
```

Comment:
Don't we have a function that's not user skippable?  Since this no longer seems to be skippable by the user, so using this function is kinda confusing.  Or should we pass `!d.Interactive` below instead of `false`?

Reply:
They are still skippable, just not managed by the user. In this case there is no way for the user to reach out this point in a non interactive session, that is why I hardcoded it to false. Now that I am writing this, it should not use the skippable prompt

Reply:
Done both comments